mlir/lib/Dialect/OpenMP/IR/OpenMPDialect.cpp

```diff
@@ -1488,6 +1520,13 @@ LogicalResult ParallelOp::verify() {
   if (failed(verifyPrivateVarList(*this)))
     return failure();
 
+  auto reductionVarsByRef = getReductionVarsByref();
+  if (reductionVarsByRef &&
+      reductionVarsByRef->size() != getReductionVars().size())
+    return emitOpError()
+           << "expected as many reduction variable by reference attributes "
+              "as reduction variables";
+
```

Comment:
Should this be in `verifyReductionVarList`?

Is there a test for this error?

Reply:
I couldn't think of a way to test for this error because we can't write text-form IR like this. I would have to create the operation with C++ like in a unit test, but I didn't know how to catch an error from a unit test.